pandas/compat/numpy/__init__.py

```diff
@@ -27,41 +27,39 @@
 _tz_regex = re.compile("[+-]0000$")
 
 
-def tz_replacer(s):
-    if isinstance(s, str):
-        if s.endswith("Z"):
-            s = s[:-1]
-        elif _tz_regex.search(s):
-            s = s[:-5]
-    return s
+def _tz_replacer(tstring):
+    if isinstance(tstring, str):
+        if tstring.endswith("Z"):
+            tstring = tstring[:-1]
+        elif _tz_regex.search(tstring):
+            tstring = tstring[:-5]
+    return tstring
 
 
-def np_datetime64_compat(s, *args, **kwargs):
+def np_datetime64_compat(tstring: str, unit: str = "ns"):
     """
     provide compat for construction of strings to numpy datetime64's with
     tz-changes in 1.11 that make '2015-01-01 09:00:00Z' show a deprecation
     warning, when need to pass '2015-01-01 09:00:00'
     """
-    s = tz_replacer(s)
-    # error: No overload variant of "datetime64" matches argument types "Any",
-    # "Tuple[Any, ...]", "Dict[str, Any]"
-    return np.datetime64(s, *args, **kwargs)  # type: ignore[call-overload]
+    tstring = _tz_replacer(tstring)
+    return np.datetime64(tstring, unit)
 
 
-def np_array_datetime64_compat(arr, *args, **kwargs):
+def np_array_datetime64_compat(arr, dtype="M8[ns]"):
```

Comment:
btw I think we can blow all of this compat code away :->

Reply:
will take a look in the next pass